packages/machines/time-picker/src/time-picker.types.ts

```diff
@@ -0,0 +1,221 @@
+import type { StateMachine as S } from "@zag-js/core"
+import { Time } from "@internationalized/date"
+import type { CommonProperties, DirectionProperty, PropTypes, RequiredBy } from "@zag-js/types"
+import type { Placement, PositioningOptions } from "@zag-js/popper"
+
+/* -----------------------------------------------------------------------------
+ * Callback details
+ * -----------------------------------------------------------------------------*/
+
+export type TimePeriod = "am" | "pm"
+
+export type TimeUnit = "hour" | "minute" | "second" | "period"
+
+export interface OpenChangeDetails {
+  open: boolean
+}
+
+export interface ValueChangeDetails {
+  value?: Time
+  valueAsString?: string
+}
+
+export interface FocusChangeDetails extends ValueChangeDetails {
+  focusedCell: {
+    value: number
+    unit: TimeUnit
+  }
+}
+
+/* -----------------------------------------------------------------------------
+ * Machine context
+ * -----------------------------------------------------------------------------*/
+
+export type ElementIds = Partial<{
+  trigger: string
+  input: string
+  positioner: string
+  content: string
+  clearTrigger: string
+  control: string
+  contentColumn: string
+}>
+
+interface PublicContext extends DirectionProperty, CommonProperties {
+  /**
+   * The locale (BCP 47 language tag) to use when formatting the time.
+   */
+  locale: string
+  /**
+   * The selected time.
+   */
+  value?: Time
+  /**
+   * Whether the timepicker is open
+   */
+  open?: boolean
+  /**
+   * Whether the datepicker open state is controlled by the user
+   */
+  "open.controlled"?: boolean
+  /**
+   * The ids of the elements in the date picker. Useful for composition.
+   */
+  ids?: ElementIds
+  /**
+   * The `name` attribute of the input element.
+   */
+  name?: string
+  /**
+   * The user provided options used to position the time picker content
+   */
+  positioning: PositioningOptions
+  /**
+   * The placeholder text of the input.
+   */
+  placeholder?: string
+  /**
+   * Whether the time picker is disabled.
+   */
+  disabled?: boolean
+  /**
+   * Whether the time picker is read-only.
+   */
+  readOnly?: boolean
+  /**
+   * The minimum time that can be selected.
+   */
+  min?: Time
+  /**
+   * The maximum time that can be selected.
+   */
+  max?: Time
+  /**
+   * The steps of each time unit.
+   */
+  steps?: { hour?: number; minute?: number; second?: number }
+  /**
+   * Whether to show the seconds.
+   */
+  withSeconds?: boolean
```

Comment:
I would suggest using a control a bit more agnostic:
```suggestion
  precision?: "hours" | "minutes" | "seconds" | "miliseconds"
```
This way, you may try and add "milliseconds" entry and also be able to use timePicker with hours only